torch/nested/_internal/nested_tensor.py

```diff
@@ -134,53 +134,29 @@ def __reduce_ex__(self, proto):
     def __tensor_flatten__(self):
         ctx = {
             "requires_grad": self.requires_grad,
-            "ragged_size": self._size[self._ragged_idx],
         }
         inner_tensors = ["_values", "_offsets"]
         if self._lengths is not None:
             inner_tensors.append("_lengths")
         return inner_tensors, ctx
 
     @staticmethod
-    def __tensor_unflatten__(inner_tensors: Dict, meta):
+    def __tensor_unflatten__(inner_tensors: Dict, meta, outer_size):
         assert len(inner_tensors) >= 2 and len(inner_tensors) <= 3
         values = inner_tensors["_values"]
         offsets = inner_tensors["_offsets"]
         lengths = inner_tensors.get("_lengths", None)
 
-        # NOTE [ Storing symbolic values as plain attributes on subclasses ]
-        #
-        # When a subclass like NestedTensor stores a "size-like" value (which
-        # can either be Symintified or not) into meta, it's responsible for:
-        #
-        #   (1) Propagating that symint during torch dispatch when performing
-        #       operations, i.e. torch dispatch plays the role of a meta kernel.
-        #
-        #   (2) Facilitating the behavior around symbolic -> non-symbolic
-        #       conversions and vice versa, see below.
-        #
-        # [ non-symbolic -> symbolic (fakification in meta_utils) ]
-        #
-        # __tensor_unflatten__ is passed symbolic dense tensors and meta from
-        # non-symbolic subclasses. In this case, the subclass is responsible for
-        # intercepting meta["ragged_size"] for example and replacing it with the
-        # symintified version.
-        #
-        # [ symbolic -> non-symbolic ]
-        #
-        # __tensor_unflatten__ is passed non-symbolic dense tensors and with
-        # meta extracted from fake subclasses. In this case the subclass gets
-        # propagated the meta["ragged_size"] which is still a symint and the
-        # subclass is responsible for making sure that the symint doesn't leak.
-        #
         # Note that we cannot simply check if is_fake(values) because
         # during aot autograd, FunctionalTensors are not fake but hold
         # symbolic sizes.
         ragged_source = offsets if lengths is None else lengths
         if has_free_symbols(ragged_source) or has_free_symbols(values):
             # Associate offsets or lengths (possibly fake, possibly functionalized)
             # with the ragged_size.
-            _tensor_symint_registry[ragged_source] = meta["ragged_size"]
+            # TODO: Utilize ragged_idx
+            ragged_size = outer_size[1]
+            _tensor_symint_registry[ragged_source] = ragged_size
```

Comment:
This is pre-existing code, but I would kind of like to know why you have to modify the registry here. Isn't ragged size guaranteed to already be associated with its source?

Reply:
It's for symbolic ragged_source -> symbolic ragged_size. The non-symbolic mapping does exist in the registry, and adding the symbolic mapping here allows the `NestedTensor` constructor to be used directly below.